contracts/token/ERC721/extensions/ERC721Consecutive.sol

```diff
@@ -141,8 +141,15 @@ abstract contract ERC721Consecutive is IERC2309, ERC721 {
         super._afterTokenTransfer(from, to, firstTokenId, batchSize);
     }
 
+    /**
+     * @dev Used to offset the first token id in {_totalConsecutiveSupply}
+     */
+    function _firstConsecutiveId() internal view virtual returns (uint96) {
```

Comment:
We need to make the `_firstConsecutiveId` doesn't affect boundaries in other functions.

For example, there's a range check of the consecutive range in [`ownerOf(...)`](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4097/files#diff-cf3d2b9b88440ac0f2ca0e32a16251bbd411cd98ea845549eb347668894212b7R59), that will be affected by the offset `_firstConsecutiveId` is adding.

We may need something like:

```solidity
if (owner != address(0) || tokenId > (type(uint96).max + _firstConsecutiveId()) || tokenId < _firstConsecutiveId()) {
  return owner;
}
```

Another example is the `_afterTokenTransfer()` function in [line 135](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4097/files#diff-cf3d2b9b88440ac0f2ca0e32a16251bbd411cd98ea845549eb347668894212b7R135), we'll need to make sure the `firstTokenId` was indeed a batch mint.

Maybe:

```solidity
if (
    to == address(0) && // if we burn
    firstTokenId >= _firstConsecutiveId() && firstTokenId - _firstConsecutiveId() < _totalConsecutiveSupply() // and the tokenId is in the batch range
    !_sequentialBurn.get(firstTokenId) // and the token was never marked as burnt
)
```

Although being honest, I don't like the amount of reads it needs

Reply:
> Although being honest, I don't like the amount of reads it needs

If the value is not in storage there is no read.

Reply:
But there are still `JUMP`s, adding overhead in a contract meant for efficiency. Even if it's a few units of gas, in a batch of 5000 we'll see the impact.

Reply:
And yes, we definitelly need to skip the lookup in `_ownerOf` if the tokenId is < offset.
The part in `_afterTokenTransfer()` is a welcome optimisation but its not as critical (just to be clear we still want it!)

Reply:
> We need to make the `_firstConsecutiveId` doesn't affect boundaries in other functions.
> 
> For example, there's a range check of the consecutive range in [`ownerOf(...)`](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4097/files#diff-cf3d2b9b88440ac0f2ca0e32a16251bbd411cd98ea845549eb347668894212b7R59), that will be affected by the offset `_firstConsecutiveId` is adding.
> 
> We may need something like:
> 
> ```solidity
> if (owner != address(0) || tokenId > (type(uint96).max + _firstConsecutiveId()) || tokenId < _firstConsecutiveId()) {
>   return owner;
> }
> ```
> 
> Another example is the `_afterTokenTransfer()` function in [line 135](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4097/files#diff-cf3d2b9b88440ac0f2ca0e32a16251bbd411cd98ea845549eb347668894212b7R135), we'll need to make sure the `firstTokenId` was indeed a batch mint.
> 
> Maybe:
> 
> ```solidity
> if (
>     to == address(0) && // if we burn
>     firstTokenId >= _firstConsecutiveId() && firstTokenId - _firstConsecutiveId() < _totalConsecutiveSupply() // and the tokenId is in the batch range
>     !_sequentialBurn.get(firstTokenId) // and the token was never marked as burnt
> )
> ```
> 
> Although being honest, I don't like the amount of reads it needs

I ended up having to cast to not get an arithmetic overflow/underflow error 
```solidity
if (owner != address(0) || tokenId > (uint256(type(uint96).max) + _firstConsecutiveId()) || tokenId < _firstConsecutiveId()) {
        return owner;
}
```